grpc/client.go

```diff
@@ -0,0 +1,363 @@
+				},
+			},
+		})
+	}
+
+	methodCfg = append(methodCfg, MethodConfig{
+		Name: []MethodName{{}}, // Empty name matches all methods
+		RetryPolicy: &RetryPolicy{
+			MaxAttempts:       cfg.MaxAttempts,
+			InitialBackoff:    cfg.InitialBackoff.String(),
+			MaxBackoff:        cfg.MaxBackoff.String(),
+			BackoffMultiplier: cfg.BackoffMultiplier,
+			RetryableStatusCodes: []string{
+				GRPCCodeCanonicalString(codes.Unavailable),
+				GRPCCodeCanonicalString(codes.DeadlineExceeded),
+				GRPCCodeCanonicalString(codes.ResourceExhausted),
+				GRPCCodeCanonicalString(codes.Aborted),
+				GRPCCodeCanonicalString(codes.Unknown),
+				GRPCCodeCanonicalString(codes.Internal),
+			},
+		},
+	})
+
+	serviceCfg := ServiceConfig{MethodConfig: methodCfg}
+
+	serviceCfgJSON, err := json.Marshal(serviceCfg)
+	if err != nil {
+		return "", fmt.Errorf("failed to marshal gRPC service config: %w", err)
+	}
+
+	return string(serviceCfgJSON), nil
+}
+
+// Conn returns the gRPC connection
+func (c *Client) Conn() *grpc.ClientConn {
+	return c.conn
+}
+
+// Close closes the gRPC connection
+func (c *Client) Close() error {
+	return c.conn.Close()
+}
+
+// GRPCError represents an error structure used in gRPC communication.
+// It contains the error code, a descriptive message, and optional details
+// providing additional context about the error.
+type GRPCError struct {
+	Code    codes.Code
+	Message string
+	Details []string
+}
+
+// Error returns a formatted string representation of the GRPCError,
+// including the error code, message, and details. The details are
+// joined into a single string for readability.
+func (e GRPCError) Error() string {
+	return fmt.Sprintf("Code: %s, Message: %s, Details: %s", e.Code.String(), e.Message, joinDetails(e.Details))
+}
+
+// RepackGRPCErrorWithDetails extracts *status.Status and formats ErrorInfo details into a single error
+func RepackGRPCErrorWithDetails(err error) error {
+	st, ok := status.FromError(err)
+	if !ok {
+		return err // Not a gRPC status error
+	}
+
+	var detailStrs []string
+	for _, d := range st.Details() {
+		if info, ok := d.(*errdetails.ErrorInfo); ok {
+			var detail string
+			if len(info.Metadata) > 0 {
+				detail += ", Metadata: {"
+				for k, v := range info.Metadata {
+					detail += fmt.Sprintf("%s: %s, ", k, v)
+				}
+				detail = strings.TrimSuffix(detail, ", ") + "}"
+			}
+
+			detailStr := fmt.Sprintf("Reason: %s, Domain: %s. %s", info.Reason, info.Domain, detail)
+			detailStrs = append(detailStrs, detailStr)
+		} else {
+			detailStrs = append(detailStrs, fmt.Sprintf("%+v", d))
+		}
+	}
+
+	return GRPCError{
+		Code:    st.Code(),
+		Message: st.Message(),
+		Details: detailStrs,
+	}
+}
+
+// joinDetails joins detail strings with a separator
+func joinDetails(details []string) string {
+	if len(details) == 0 {
+		return noneStr
+	}
+	return fmt.Sprintf("[%s]", strings.Join(details, ";"))
+}
+
+// GRPCCodeCanonicalString returns the canonical service config string for a gRPC status code.
```

Comment:
It's not clear what this function does. Can you add an example to the function comment?

Reply:
Fixed https://github.com/agglayer/aggkit/pull/524/commits/33643b1e14622bfce3384fc5ee68bd764d16365f